soweego/linker/classifiers.py

```diff
@@ -95,7 +96,109 @@ def prob(self, feature_vectors: pd.DataFrame) -> pd.DataFrame:
         # so we return the second column
         classifications = self.kernel.predict_proba(feature_vectors)[:, 1]
 
-        return pd.DataFrame(classifications, index=feature_vectors.index)
+        return pd.Series(classifications, index=feature_vectors.index)
+
+    def __repr__(self):
+        return f'{self.kernel}'
+
+
+class RandomForest(SKLearnAdapter, BaseClassifier):
+    """A Random Forest classifier.
+
+    This class implements :class:`sklearn.ensemble.RandomForestClassifier`.
+
+    It fits multiple decision trees on sub-samples of the dataset and
+    averages the result to get more accuracy and reduce over-fitting.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(RandomForest, self).__init__()
+
+        kwargs = {**constants.RANDOM_FOREST_PARAMS, **kwargs}
+        self.kernel = RandomForestClassifier(*args, **kwargs)
+
+    def prob(self, feature_vectors: pd.DataFrame) -> pd.DataFrame:
+        """Classify record pairs and include the probability score
+        of being a match.
+
+        :param feature_vectors: a :class:`DataFrame <pandas.DataFrame>`
+          computed via record pairs comparison. This should be
+          :meth:`recordlinkage.Compare.compute` output.
+          See :func:`extract_features() <soweego.linker.workflow.extract_features>`
+          for more details
+        :return: the classification results
+        """
+
+        match_class = self.kernel.classes_[1]
+
+        # Invalid class label
+        assert match_class == 1, (
+            f'Invalid match class label: {match_class}.'
+            'sklearn.ensemble.RandomForestClassifier.predict_proba() expects the second class '
+            'in the trained model to be 1'
+        )
+
+        # in the result, rows are classifications and columns are classes.
+        # We are in a binary setting, so 2 classes:
+        # `0` for non-matches, `1` for matches.
+        # We only need the probability of being a match,
+        # so we return the second column
+        classifications = self.kernel.predict_proba(feature_vectors)[:, 1]
+
+        return pd.Series(classifications, index=feature_vectors.index)
+
+    def __repr__(self):
+        return f'{self.kernel}'
+
+
+class VoteClassifier(SKLearnAdapter, BaseClassifier):
```

Comment:
Suggestion: why don't we call this object EnsembleClassifier?

Reply:
For two reasons:

1. To prevent confusion, since in the `implement-ensembles` branch I've implemented other ensembling techniques (`gate` and `stack`), and it might be misleading having one of them named `EnsembleClassifer`
2. `VoteClassifier` is the actual name of this technique of mixing results from different classifiers.